microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/java/JavaFeatureDefinition.java

```diff
@@ -27,62 +29,65 @@
  * 
  * <ul>
  * <li>{@link IJavaHoverParticipant}.</li>
+ * <li>{@link IJavaCodeLensParticipant}.</li>
  * <li>{@link IJavaDiagnosticsParticipant}.</li>
  * </ul>
  *
  */
-public class JavaFeatureDefinition implements IJavaHoverParticipant, IJavaDiagnosticsParticipant {
+public class JavaFeatureDefinition
+		implements IJavaHoverParticipant, IJavaCodeLensParticipant, IJavaDiagnosticsParticipant {
 	private static final Logger LOGGER = Logger.getLogger(JavaFeatureDefinition.class.getName());
 
-	private final IJavaHoverParticipant hoverParticipant;
+	private final IJavaCodeLensParticipant codeLensParticipant;
 	private final IJavaDiagnosticsParticipant diagnosticsParticipant;
+	private final IJavaHoverParticipant hoverParticipant;
 
-	public JavaFeatureDefinition(IJavaHoverParticipant hoverParticipant,
-			IJavaDiagnosticsParticipant diagnosticsParticipant) {
-		this.hoverParticipant = hoverParticipant;
+	public JavaFeatureDefinition(IJavaCodeLensParticipant codeLensParticipant,
```

Comment:
What do you think about using constructor overloading instead? So that we can create instances  `new JavaFeatureDefinition(codeLensParticipant)` instead of `new JavaFeatureDefinition(codeLensParticipant, null, null)`?